### PR TITLE
Strip "data-test-" attributes from component and helper invocations

### PR DESCRIPTION
**Pull request by Turbo87**
This automatically removes `data-test-` prefixed attributes from component/helper invocations in templates like `{{some-component data-test-name="foobar"}}`. This is roughly equal to what we've already been doing on vanilla HTML tags like `<div data-test-name="foobar">`.

### CODE REVIEW

File: tests/dummy/app/templates/components/print-test-attributes.hbs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<div class="attr">{{data-test-attribute}}</div>
```
Comment (marcoow) [inline tests/dummy/app/templates/components/print-test-attributes.hbs]: This is an example of using a `data-test-*` attribute in the wrong place actually. It should be stripped but should actually never appear here in the first place.

File: tests/integration/strip-data-test-attributes-from-components-test.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import hbs from "htmlbars-inline-precompile";
+
+import config from 'dummy/config/environment';
+
+moduleForComponent('print-test-attributes', 'StripTestSelectorsTransform plugin', {
+  integration: true
+});
+
+if (config.stripTestSelectors) {
+
```
Comment (marcoow) [inline tests/integration/strip-data-test-attributes-from-components-test.js]: This needs a test that while

```hbs
{{my-component data-test-attr='value'}}
```

becomes

```html
<div class="ember-view" data-test-attr="value">
  …
</div>
```

in environments that we don't strip attributes for it becomes

```html
<div class="ember-view">
  …
</div>
```

when `data-test-*` attributes are stripped.

File: tests/integration/strip-data-test-attributes-from-components-test.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    this.render(hbs`{{print-test-attributes data-test="foo"}}`);
+
+    assert.equal(this.$('.data-test').text(), 'foo', 'the .data-test div is empty');
+  });
+
+  test('it leaves other data attributes untouched on components', function (assert) {
+    this.render(hbs`{{print-test-attributes data-non-test="foo"}}`);
+
+    assert.equal(this.$('.non-test').text(), 'foo', 'the .non-test div is not empty');
+  });
```
Comment (marcoow) [inline tests/integration/strip-data-test-attributes-from-components-test.js]: These cases actually all test edge cases. I guess that's fine but they all assume wrong usages of `data-test-*` attributes.

**Comment by marcoow** [comment]
This should also test that `data-test-*` attributes in a component's context are stripped, e.g.

```js
export default Component.extend({
  'data-test-my-attr': computed(function() {
    return 'whatever';
  })
```

File: tests/dummy/app/templates/components/print-test-attributes.hbs (hunk 1 above)
Comment (pangratz) [inline tests/dummy/app/templates/components/print-test-attributes.hbs]: This is only used for testing purposes to see that the `data-test-attribute` is correctly stripped.

File: tests/dummy/app/templates/components/print-test-attributes.hbs [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+<div class="second">{{data-test-second}}</div>
+<div class="non-test">{{data-non-test}}</div>
+<div class="data-test">{{data-test}}</div>
```
Comment (pangratz) [inline tests/dummy/app/templates/components/print-test-attributes.hbs]: I think the class names in here should be named consistently to reflect the corresponding attribute. I would suggest the class name to be the same name as the attribute, so we have `.data-test-attribute`, `.data-test-second`, `.data-non-test` and `.data-test`.

**Comment by Turbo87** [comment]
After a short voice conversation we've resolved a small misunderstanding about the purpose of this PR and the tests related to the `print-test-attributes` component. 

tl;dr this PR is focused **only** on the stripping of `data-test-` attributes from component/helper invocations and the corresponding tests for that feature

File: tests/dummy/app/templates/components/print-test-attributes.hbs (hunk 4 above)
Comment (Turbo87) [inline tests/dummy/app/templates/components/print-test-attributes.hbs]: @pangratz good idea. done.
